### DISCUSSION TITLE
meta mask

### DISCUSSION THREAD

**Discussion by 963reSilient** [Cyfrin/foundry-full-course-cu#476] [Q&A]
Is anyone else concerned about meta masks' ability to "read and change all your data on all websites" ?

**Comment by Niferu** [comment]
Yea thats really good question. I know that all wallet extensions have this permission and good solution is to use them on separate browser where you do not use your credentials for banks or other sensitive accounts. But I'm not cyber security expert, so maybe @PatrickAlphaC has any thoughts  about it?

**Reply by PatrickAlphaC** [reply]
Here is a great response on this:

https://ethereum.stackexchange.com/questions/39954/does-metamask-store-private-key-on-server-or-anywhere-else

But... keep in mind, they do push changes, and it's still code at the end of the day...

**Reply by Niferu** [reply]
Indeed they cover topic a little bit, but that response did not include reading passwords, logins etc. on off chain websites.

**Reply by 963reSilient** [reply]
Thank You Everyone,
I decided to download Brave and do all of my studying from this platform. I've downloaded everything I need, and  decided to start the course over from the beginning. No harm in reinforcing information! Thanks again.
